constraints.txt

```diff
@@ -0,0 +1 @@
+epicscorelibs==7.0.7.99.1.1a3
```

Comment:
I don't think this is being used currently?

In the workflow I think you need to export `PIP_CONTRAINT` as an env var, pointing at this file.

(The reason we do it as an env var rather than command line flag is so that it applies to subenvironments used for building too)

Reply:
will modify our reusable workflow to set it, i think setting it in this workflow won't propagate to the reused one: https://docs.github.com/en/actions/sharing-automations/reusing-workflows#limitations

Reply:
Mostly matters for running the tests, not too fussed about ruff/pyright one.